cli/module_build.go

```diff
@@ -395,8 +391,10 @@ func jobStatusFromProto(s buildpb.JobStatus) jobStatus {
 
 // ReloadModuleAction builds a module, configures it on a robot, and starts or restarts it.
 func ReloadModuleAction(c *cli.Context) error {
-	if !c.Bool(moduleBuildRestartOnly) {
-		return fmt.Errorf("only %s mode is supported for now", moduleBuildRestartOnly)
+	if len(c.String(partFlag)) > 0 && !c.Bool(moduleBuildRestartOnly) {
+		// todo: remove this warning after remote reloading
+		warningf(c.App.Writer,
+			"You have passed in a part ID -- if it's for a remote device and your module isn't in the expected path, this will fail")
 	}
 	partID, err := resolvePartID(c, "/etc/viam.json")
```

Comment:
When a lot of people are testing modules locally, I suspect that they are using a viam.json that is not at `/etc/viam.json` (ex: all mac users do not have their configs there).

I think that the `partFlag` should be supported for local configs as well (or there should be a way to redirect the cli to use a different viam-config, but I'd rather just pass the id as a cli arg)

Reply:
it's supported, it just warns

we could distinguish the 'local part id' case from the remote case if we had some way to test where the part ID is running -- but this warning goes away in v2 of the feature anyway

Reply:
Misread `resolvePartID` -- it also checks c.String(partFlag). Talked offline. Moving that out of the helper func

Reply:
1e6e9b4b helper function takes flag directly now